cmd/wallet-sdk-gomobile/openid4vp/openid4vp.go

```diff
@@ -8,36 +8,81 @@ SPDX-License-Identifier: Apache-2.0
 package openid4vp
 
 import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jwt"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/presexch"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+
 	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
-	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/linkeddomains"
+	goapi "github.com/trustbloc/wallet-sdk/pkg/api"
+	"github.com/trustbloc/wallet-sdk/pkg/common"
+	"github.com/trustbloc/wallet-sdk/pkg/openid4vp"
 )
 
-// Instance is used to help with OpenID4VP operations (TODO: Implement).
-type Instance struct {
-	authorizationRequest []byte
-	keyHandleReader      api.KeyReader
-	didResolver          api.DIDResolver
-	linkedDomains        *linkeddomains.LD
+type goAPIOpenID4VP interface {
+	GetQuery() (*presexch.PresentationDefinition, error)
+	PresentCredential(presentation *verifiable.Presentation, jwtSigner goapi.JWTSigner) error
 }
 
-// NewInstance returns a new OpenID4VP Instance.
-func NewInstance(authorizationRequest []byte, keyHandleReader api.KeyReader,
-	didResolver api.DIDResolver, linkedDomains *linkeddomains.LD,
-) (*Instance, error) {
-	return &Instance{
-		authorizationRequest: authorizationRequest,
-		keyHandleReader:      keyHandleReader,
-		didResolver:          didResolver,
-		linkedDomains:        linkedDomains,
-	}, nil
+// Interaction represents a single OpenID4VP interaction between a wallet and a verifier. The methods defined on this
+// object are used to help guide the calling code through the OpenID4VP flow.
+type Interaction struct {
+	keyHandleReader  api.KeyReader
+	crypto           api.Crypto
+	ldDocumentLoader api.LDDocumentLoader
+	goAPIOpenID4VP   goAPIOpenID4VP
 }
 
-// GetQuery does something (TODO: Implement).
-func (o *Instance) GetQuery() ([]byte, error) {
-	return []byte("Example"), nil
+// NewInteraction creates a new OpenID4VP Interaction.
+// The methods defined on this object are used to help guide the calling code through the OpenID4CI flow.
+func NewInteraction(authorizationRequest []byte, keyHandleReader api.KeyReader, crypto api.Crypto,
```

Comment:
Can we keep `authorizationRequest` as string instead of []byte ? I see this finally being casted to string.

Reply:
Where is this authorization request coming from? I assume it starts as a string so we should keep it as a string, it helps with inspection (logging, debugging, etc). Or if it's a predefined JSON object, we can also pass that.

Reply:
@skflorin This is the redirectURL from QR code.

Reply:
Then it should be a string

Reply:
Oh yeah, it gets converted to a string in the Go SDK layer (in the `pkg` folder). It should be a string in the Go SDK layer too

Reply:
Not: It's used [here](https://github.com/trustbloc/wallet-sdk/pull/66/files#diff-b0b99fe81cda029548c9aadd83389fddbbdfdfe795a9905711c7cebb453a532eR104) to call retrieve JWT query.

Reply:
What I mean was for serialized objects in JSON format, are they usually strings or []byte on the mobile side @skflorin? In Go, they're []byte. For the other API methods I added, I used []byte but I'm not sure if I should use strings instead to make the mobile side easier

Reply:
(When I say the []byte input, I assumed it was JSON)

Reply:
@DRK3 strings are much easier to work with on mobile. byte[] is used for low level crypto or similar low level things that need to work on bytes.